src/server/pfs/server/master.go

```diff
@@ -70,21 +86,32 @@ func (d *driver) master(ctx context.Context) {
 }
 
 func (d *driver) finishCommits(ctx context.Context) error {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
The better pattern imo is to have the defer cancel immediately after it is created to ensure it is canceled upon exit. Having code sandwiched between its creation and the deferred cancel can result with issues over time if more logic is inserted between the two.